spark-operator/src/main/java/org/apache/spark/k8s/operator/config/SparkOperatorConf.java

```diff
@@ -0,0 +1,463 @@
+                  + "disabled, operator startup will ignore recoverable errors, "
+                  + "caused for example by RBAC issues and will retry "
+                  + "periodically.")
+          .typeParameterClass(Boolean.class)
+          .defaultValue(false)
+          .build();
+
+  public static final ConfigOption<Integer> RECONCILER_TERMINATION_TIMEOUT_SECONDS =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.reconciler.terminationTimeoutSeconds")
+          .enableDynamicOverride(false)
+          .description(
+              "Grace period for operator shutdown before reconciliation threads are killed.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(30)
+          .build();
+
+  public static final ConfigOption<Integer> RECONCILER_PARALLELISM =
+      ConfigOption.<Integer>builder()
+          .key("spark.kubernetes.operator.reconciler.parallelism")
+          .enableDynamicOverride(false)
+          .description(
+              "Thread pool size for Spark Operator reconcilers. Unbounded pool would be used if "
+                  + "set to non-positive number.")
+          .typeParameterClass(Integer.class)
+          .defaultValue(50)
+          .build();
+
+  public static final ConfigOption<Long> RECONCILER_FOREGROUND_REQUEST_TIMEOUT_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.reconciler.foregroundRequestTimeoutSeconds")
+          .enableDynamicOverride(true)
+          .description(
+              "Timeout (in seconds) to for requests made to API server. this "
+                  + "applies only to foreground requests.")
+          .typeParameterClass(Long.class)
+          .defaultValue(30L)
+          .build();
+
+  public static final ConfigOption<Long> SPARK_APP_RECONCILE_INTERVAL_SECONDS =
+      ConfigOption.<Long>builder()
+          .key("spark.kubernetes.operator.reconciler.intervalSeconds")
+          .enableDynamicOverride(true)
+          .description(
+              "Interval (in seconds, non-negative) to reconcile Spark applications. Note that "
+                  + "reconciliation is always expected to be triggered when app spec / status is "
+                  + "updated. This interval controls the reconcile behavior of operator "
+                  + "reconciliation even when there's no update on SparkApplication, e.g. to "
+                  + "determine whether a hanging app needs to be proactively terminated. Thus "
+                  + "this is recommended to set to above 2 min to avoid unnecessary no-op "
+                  + "reconciliation.")
+          .typeParameterClass(Long.class)
+          .defaultValue(120L)
+          .build();
+
+  public static final ConfigOption<Boolean> TRIM_ATTEMPT_STATE_TRANSITION_HISTORY =
+      ConfigOption.<Boolean>builder()
+          .key("spark.kubernetes.operator.reconciler.trimStateTransitionHistoryEnabled")
```

Comment:
To use `Enabled` here and `spark.kubernetes.operator.metrics.clientMetricsEnabled`, we need to change the following for consistency.
```
- spark.kubernetes.operator.terminateOnInformerFailure
+ spark.kubernetes.operator.terminateOnInformerFailureEnabled
```